tests/std/tests/P0980R1_constexpr_strings/test.cpp

```diff
@@ -0,0 +1,1758 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma warning(disable : 4389) // signed/unsigned mismatch in arithmetic
+
+#ifdef __clang__
+#pragma clang diagnostic ignored "-Wsign-compare"
+#endif // __clang__
+
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <string>
+#include <string_view>
+
+using namespace std;
+

```

Comment:
This test is failing for Clang in the MSVC-internal repo; note that `MSVC_INTERNAL_TESTING` is defined for all compilers. I think we should probably guard the entire test with `#ifdef __cpp_lib_constexpr_string`, which will prevent it from running for Clang until we can activate it for Clang 12.

Reply:
What sort of failures are you seeing?  I believe the intent was for Clang to not evaluate the `static_assert(test_*)` calls (as those are only activated for `__cpp_lib_constexpr_string`, which shouldn't be defined for Clang).  Are the failures caused by the plain `constexpr`s on a few helper functions in the test?  

If so, I wonder if changing those to `_CONSTEXPR20_CONTAINER` resolve the issue?  Then we would maintain the runtime tests for Clang (though given the simplicity of the actual tests exercising each function, I don't know that the runtime tests are providing much additional test coverage).

Reply:
Now that I'm looking at it again, the first clang-cl error is complaining about the type mismatch, so this may not be a unique problem. I'll see if fixing the other issues resolves this - thanks for explaining the intent.

Reply:
Yep, Clang appears to be passing after fixing the other issues.